src/abstract-interpretation/handler/binop/operators.ts

```diff
@@ -1,49 +1,48 @@
-import {guard} from '../../../util/assert'
-import {BinOpOperators} from './binop'
-import {addDomains, Domain, narrowDomain, NarrowKind, subtractDomains} from '../../domain'
-import {AINode, AINodeStore} from '../../ainode'
+import { addDomains, Domain, narrowDomain, NarrowKind, subtractDomains } from '../../domain'
+import type { BinOpOperators } from './binop'
+import { AINode, AINodeStore } from '../../ainode'
+import type { ParentInformation } from '../../../r-bridge/lang-4.x/ast/model/processing/decorate'
+import type { RBinaryOp } from '../../../r-bridge/lang-4.x/ast/model/nodes/r-binary-op'
+
+const calculateComparisonDomains = (lhs: AINode, rhs: AINode, node: RBinaryOp<ParentInformation>, narrowKind: NarrowKind, idSuffix = ''): AINode[] => {
+	const lhsNarrowed = narrowDomain(lhs.domain, rhs.domain, narrowKind)
+	const rhsNarrowed = narrowDomain(rhs.domain, lhs.domain, narrowKind ^ 0b110 /* flip < and > but not = */)
+	const isConditionTrue = lhsNarrowed.isBottom() && rhsNarrowed.isBottom()
+	return [
+		new AINode(isConditionTrue ? Domain.bottom() : Domain.top(), node, node.info.id, node.info.id + idSuffix),
+		new AINode(lhsNarrowed, node, node.info.id, lhs.nodeId + idSuffix),
+		new AINode(rhsNarrowed, node, node.info.id, rhs.nodeId + idSuffix),
+	]
+}
 
 export const operators: BinOpOperators = {
-	'assignment': (_, rhs, node) => {
-		return AINodeStore.from(new AINode(rhs.domain, node.lhs, node.info.id))
+	'<-': (lhs, rhs, node) => {
```

Comment:
Such mappers should slowly be trasnitioned to the `built-in` environment schema of handlers.

Reply:
Yep, I'm already on it